genesyscloud/resource_genesyscloud_widget_deployment.go

```diff
@@ -0,0 +1,405 @@
+
+func parseSdkClientConfigData(d *schema.ResourceData) (webchatSkin *string, authenticationUrl *string) {
+	clientConfigSet := d.Get("client_config").(*schema.Set)
+
+	if clientConfigSet != nil && len(clientConfigSet.List()) > 0 {
+		clientConfig := clientConfigSet.List()[0].(map[string]interface{})
+		fields := make(map[string]string)
+
+		for k, v := range clientConfig {
+			fields[k] = v.(string)
+		}
+
+		webchatSkin := fields["webchat_skin"]
+		authUrl := fields["authentication_url"]
+		return &webchatSkin, &authUrl
+	}
+	return nil, nil
+}
+
+func validateAuthURL(authUrl interface{}, _ cty.Path) diag.Diagnostics {
+	authUrlString := authUrl.(string)
+	u, err := url.Parse(authUrlString)
+	if err != nil {
+		return diag.Errorf("Authorization url %s provided is not a valid URL", authUrlString)
+	}
+
+	if u.Scheme == "" || u.Host == "" {
+		log.Printf("Scheme: %s", u.Scheme)
+		log.Printf("Host: %s", u.Host)
+		return diag.Errorf("Authorization url %s provided is not valid url", authUrlString)
+	}
+
+	if u.Scheme != HTTPSPROTOCOL {
+		return diag.Errorf("Authorization url %s provided must begin with https", authUrlString)
+	}
+
+	return nil
+}
+
+func buildSDKClientConfig(clientType string, d *schema.ResourceData) (*platformclientv2.Widgetclientconfig, error) {
+	widgetClientConfig := &platformclientv2.Widgetclientconfig{}
+	clientConfig := d.Get("client_config").(*schema.Set)
+
+	if (clientType == V1 || clientType == V1HTTP) && clientConfig.Len() == 0 {
+		return nil, fmt.Errorf("V1 and v1-http widget configurations must have a client_config defined. ")
+	}
+
+	if clientType == V1 {
+		v1Client := &platformclientv2.Widgetclientconfigv1{}
+
+		v1Client.WebChatSkin, v1Client.AuthenticationUrl = parseSdkClientConfigData(d)
+
+		widgetClientConfig.V1 = v1Client
+	}
+
+	if clientType == V1HTTP {
+		v1HttpClient := &platformclientv2.Widgetclientconfigv1http{}
+		v1HttpClient.WebChatSkin, v1HttpClient.AuthenticationUrl = parseSdkClientConfigData(d)
+		widgetClientConfig.V1Http = v1HttpClient
+	}
+
+	return widgetClientConfig, nil
+}
+
+func widgetDeploymentExporter() *ResourceExporter {
+	return &ResourceExporter{
+		GetResourcesFunc: getAllWithPooledClient(getAllWidgetDeployments),
+		RefAttrs: map[string]*RefAttrSettings{
+			"flow_id": {},
+		},
+	}
+}
+
+func resourceWidgetDeployment() *schema.Resource {
+	return &schema.Resource{
+		Description: "Genesys Cloud Widget Deployment",
+
+		CreateContext: createWithPooledClient(createWidgetDeployment),
+		ReadContext:   readWithPooledClient(readWidgetDeployment),
+		UpdateContext: updateWithPooledClient(updateWidgetDeployment),
+		DeleteContext: deleteWithPooledClient(deleteWidgetDeployment),
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+		SchemaVersion: 1,
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "Name of the Widget Deployment.",
+				Type:        schema.TypeString,
+				Required:    true,
+				ForceNew:    true,
```

Comment:
not sure if ForceNew is necessary here. It's usually for a property that can't be updated. I removed it and updated the name in a test without issue

Reply:
Fixed